sdk/iot/hub/tests/cmocka/test_az_iot_hub_client_c2d.c

```diff
@@ -0,0 +1,244 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_az_iot_hub_client.h"
+#include <az_iot_hub_client.h>
+#include <az_span.h>
+
+#include <az_precondition_internal.h>
+#include <az_precondition.h>
+
+#include <setjmp.h>
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdint.h>
+
+#include <cmocka.h>
+#include <az_test_precondition.h>
+
+#define TEST_DEVICE_ID_STR "my_device"
+#define TEST_DEVICE_HOSTNAME_STR "myiothub.azure-devices.net"
+#define TEST_MODULE_ID "my_module_id"
+
+static const az_span test_device_hostname = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_HOSTNAME_STR);
+static const az_span test_device_id = AZ_SPAN_LITERAL_FROM_STR(TEST_DEVICE_ID_STR);
+static uint8_t g_test_correct_subscribe_topic[] = {'d','e','v','i','c','e','s','/','m','y','_','d','e','v','i','c','e','/','m','e','s','s','a','g','e','s','/','d','e','v','i','c','e','b','o','u','n','d','/', '#'};
```

Comment:
Might be easier but I have these correct ones as `char[]` and then use the `assert_memory_equal` function to make sure they are equal.

Reply:
Changed